discord/state.py

```diff
@@ -831,6 +831,12 @@ def parse_channel_delete(self, data: gw.ChannelDeleteEvent) -> None:
                             guild._scheduled_events.pop(s.id)
                             self.dispatch('scheduled_event_delete', s)
 
+                threads = guild._remove_threads_by_channel(channel_id)
+
+                for thread in threads:
+                    self.dispatch('thread_delete', thread)
+                    self.dispatch('raw_thread_delete', RawThreadDeleteEvent._from_thread(thread))
```

Comment:
I get why this is here but this would make it the only time (to my memory?) that a raw event doesn't correspond to its own Discord event I think.

Reply:
It is the only one yeah. But I think it'd be pretty unfortunate to not receive the raw thread delete event only in certain situations and needing to work around that.